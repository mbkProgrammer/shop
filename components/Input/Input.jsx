import { useTheme } from '@emotion/react';
import { css } from '@emotion/css';

const Input = ({
  placeholder,
  size,
  type,
  styles = '',
  onChange,
  error = false,
  errorMassage = 'not valid',
  ...props
}) => {
  const theme = useTheme();

  return (
    <>
      <input
        data-testid="input--test"
        placeholder={placeholder}
        type={type}
        onChange={onChange}
        {...props}
        className={`input--${size} ${error ? 'input--error' : ''} ${css`
          height: 40px;
          margin: 5px;
          font-size: 1em;
          font-weight: 400;
          line-height: 25px;
          border: 0.5px solid ${theme.colors.primary};
          border-radius: 5px;
          width: 445px;
          transition: 0.3s;
          background: ${theme.colors.secondary};
          &:focus {
            outline: none;
            border: 2px solid ${theme.colors.primary};
          }
          &:hover {
            border: 2px solid ${theme.colors.primary};
            background: none;
          }
          &:disabled {
            background: rgba(100, 100, 100, 0.6);
          }
          &.input--error {
            border: 2px solid ${theme.colors.error};
          }
          &.input--big {
            width: 445px;
            ${styles}
          }
          &.input--small {
            width: 140px;
            ${styles}
          }
          &.input--medium {
            width: 216px;
            ${styles}
          }
          @media (max-width: 480px) {
            &.input--big,
            &.input--small,
            &.input--medium {
              width: 100%;
            }
          }
        `}`}
      />
      {error ? (
        <p
          className={css`
          font-size: 14px;
          color: ${theme.colors.error};
          text-align: left;
        `}
        >
          {errorMassage}
        </p>
      ) : (
        ''
      )}
    </>
  );
};

export default Input;
