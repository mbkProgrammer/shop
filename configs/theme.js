const theme = {
  colors: {
    primary: '#56B280',
    secondary: '#F5FBF8',
    text: '#222322',
    background: '#EAF6F0',
  },
  typography: {
    body: {
      fontWeight: '300',
      fontSize: '1rem',
    },
    h1: {
      fontWeight: '700',
      fontSize: '2.4rem',
    },
    h2: {
      fontWeight: '700',
      fontSize: '2.2rem',
    },
    h3: {
      fontWeight: '700',
      fontSize: '2rem',
    },
    h4: {
      fontWeight: '600',
      fontSize: '1.8rem',
    },
    h5: {
      fontWeight: '600',
      fontSize: '1.6rem',
    },
    h6: {
      fontWeight: '500',
      fontSize: '1.4rem',
    },
  },
};

export default theme;
