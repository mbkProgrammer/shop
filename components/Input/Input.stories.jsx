import React from 'react';
import Input from './Input';

export default { title: 'components/Input' };
export const InputSmall = () => (<Input size="small" placeholder="Small input" type="text" />);
export const InputMedium = () => (<Input size="medium" placeholder="Small input" type="text" />);
export const InputBig = () => (<Input size="big" placeholder="Small input" type="text" />);
