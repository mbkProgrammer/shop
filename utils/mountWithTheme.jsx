import { ThemeProvider } from '@emotion/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../configs/Store';
import theme from '../configs/theme';

const mountWithTheme = (children) => renderer.create(
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </Provider>,
);

export default mountWithTheme;
