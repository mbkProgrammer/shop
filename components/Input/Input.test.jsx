import { ThemeProvider } from '@emotion/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import Input from './Input';
import theme from '../../configs/theme';
import mountWithTheme from '../../utils/mountWithTheme';

test('Input smaple', () => {
  const component = mountWithTheme(
    <Input placeholder="Input test" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Input smaple onChange', () => {
  let variable = '';
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <Input varaint="contained" onClick={() => { variable = 1; }}>Facebook</Input>
    </ThemeProvider>,
  );

  fireEvent.click(getByText(/Facebook/));
  expect(variable).toBe(2);
});
