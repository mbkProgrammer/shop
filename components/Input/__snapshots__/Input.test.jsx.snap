// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input smaple 1`] = `
Array [
  <input
    className="input--undefined  css-18urk4r"
    data-testid="input--test"
    placeholder="Input test"
  />,
  "",
]
`;

exports[`Input smaple size big 1`] = `
Array [
  <input
    className="input--big  css-18urk4r"
    data-testid="input--test"
    placeholder="Input test"
  />,
  "",
]
`;

exports[`Input smaple type password 1`] = `
Array [
  <input
    className="input--undefined  css-18urk4r"
    data-testid="input--test"
    placeholder="Input test"
    type="password"
  />,
  "",
]
`;
